rdrf/rdrf/admin.py

```diff
@@ -130,104 +174,142 @@ def export_registry(registry, request):
+            response = HttpResponse(FileWrapper(yaml_file), content_type='text/yaml')
+            yaml_export_filename = "export_%s_%s" % (export_time, yaml_export_filename)
+            response['Content-Disposition'] = 'attachment; filename="%s"' % yaml_export_filename
 
-        for registry in registrys:
-            yaml_data = export_registry(registry, request)
-            if yaml_data is None:
-                return
+            return response
+        else:
+            import zipfile
+            zipped_file = io.BytesIO()
+            zf = zipfile.ZipFile(zipped_file, mode='w', compression=zipfile.ZIP_DEFLATED)
 
-            zf.writestr(registry.code + '.yaml', yaml_data)
+            for registry in registries:
+                yaml_data = self.export_registry(registry, request)
+                if yaml_data is None:
+                    return
 
-        zf.close()
-        zippedfile.flush()
-        zippedfile.seek(0)
+                zf.writestr(registry.code + '.yaml', yaml_data)
 
-        response = HttpResponse(FileWrapper(zippedfile), content_type='application/zip')
-        name = "export_" + export_time + "_" + \
-            reduce(lambda x, y: x + '_and_' + y, [r.code for r in registrys]) + ".zip"
-        response['Content-Disposition'] = 'attachment; filename="%s"' % name
+            zf.close()
+            zipped_file.flush()
+            zipped_file.seek(0)
 
-        return response
+            response = HttpResponse(FileWrapper(zipped_file), content_type='application/zip')
+            name = "export_" + export_time + "_" + \
+                   reduce(lambda x, y: x + '_and_' + y, [r.code for r in registries]) + ".zip"
+            response['Content-Disposition'] = 'attachment; filename="%s"' % name
 
+            return response
+    export_registry_action.short_description = _("Export")
 
-export_registry_action.short_description = "Export"
+    def generate_questionnaire_action(self, request, registry_models_selected):
+        for registry in registry_models_selected:
+            registry.generate_questionnaire()
 
+    generate_questionnaire_action.short_description = _("Generate Questionnaire")
 
-def design_registry_action(modeladmin, request, registry_models_selected):
-    if len(registry_models_selected) != 1:
-        return
-    else:
-        registry = [r for r in registry_models_selected][0]
-        return HttpResponseRedirect(reverse('rdrf_designer', args=(registry.pk,)))
+    def enable_registration_action(self, request, registry_models_selected):
+        unchanged = []
 
+        for registry in registry_models_selected:
+            if registry.has_feature(RegistryFeatures.REGISTRATION):
+                unchanged.append(registry.code)
 
-design_registry_action.short_description = _("Design")
+            registry.add_feature(RegistryFeatures.REGISTRATION)
+            registry.save()
 
+            messages.success(request, f"Registration enabled for '{registry.code}'")
```

Comment:
For all messages can we use the same display format for the registries as in the table below?
ie. Human-friendly name and code in parens.